### PR TITLE
cleanup manifestworks when klusterlet is deleting

### PR DESCRIPTION
**Pull request by skeeey**

### CODE REVIEW

File: pkg/controllers/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go [hunk 1]
```diff
@@ -289,21 +313,33 @@ func (n *klusterletCleanupController) removeKlusterletFinalizers(ctx context.Con
...
+	unavailable, err := helpers.IsClusterUnavailable(ctx, n.controlplaneClusterClient, config.ClusterName)
+	if err != nil {
 		return err
 	}
 
-	if err := n.controlplaneClusterClient.ClusterV1().ManagedClusters().Delete(
-		ctx, config.ClusterName, metav1.DeleteOptions{}); err != nil {
+	if unavailable {
+		// managed cluster is unavailable, force delete all manifestworks
+		return helpers.DeleteAllManifestWorks(ctx, n.controlplaneWorkClient, works.Items, true)
```
Comment (clyang82) [inline pkg/controllers/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go:334]: Is it align with the current ACM behaviour? I remembered if the managed cluster is unavailable, the manifestwork won't be deleted forcely. correct?

Comment (clyang82) [reply pkg/controllers/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go:334]: maybe plus the deletion timestamp exists in the managed cluster?

Comment (skeeey) [reply pkg/controllers/klusterlet/controllers/klusterletcontroller/klusterlet_cleanup_controller.go:334]: no, if the managed cluster is unavailable, we will force delete the manifest work, when the klusterlet is deleting, if the managed cluster is  unavailable, that means we cannot do anything on the managed cluster, we can only delete the works forcely

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**